tests/gctests.js

```diff
@@ -148,7 +209,80 @@ tests.push(function() {
   next();
 });
 
-// TODO: Test J2ME.newArray with a primitive array (similar to the _gcMallocAtomic test)
+// Make sure values inside the gcMallocAtomic allocated area aren't considered pointers
+tests.push(function() {
+  var uncollectableAddr = ASM._gcMallocUncollectable(4);
+  var addr = J2ME.newIntArray(4);
+  i32[uncollectableAddr >> 2] = addr;
+  var objAddr = J2ME.allocObject(CLASSES.java_lang_Object.klass);
+  i32[addr + J2ME.Constants.ARRAY_HDR_SIZE >> 2] = objAddr;
+  ASM._forceCollection();
+  is(freedAddr, objAddr, "Values in an area allocated via newArray (primitive) aren't considered references");
+
+  ASM._gcFree(uncollectableAddr);
+
+  ASM._forceCollection();
+  is(freedAddr, addr, "Object not referenced by anyone anymore is freed");
+
+  ASM._forceCollection();
+  is(freedAddr, uncollectableAddr, "Finalizer is called for a freed object");
+
+  next();
+});
+
+tests.push(function() {
+  var zeroedOut = true;
+  for (var i = 0; i < 1000; i++) {
+    var addr = ASM._gcMallocAtomic(8);
+    for (var j = 0; j < 8; j += 4) {
+      if (i32[addr + j >> 2] != 0) {
+        zeroedOut = false;
+        break;
+      }
+    }
+
+    if (!zeroedOut) {
+      break;
+    }
+
+    for (var j = 0; j < 8; j += 4) {
+      i32[addr + j >> 2] = 0xDEADBEEF;
+    }
+
+    ASM._forceCollection();
+  }
+
+  ok(!zeroedOut, "gcMallocAtomic doesn't zero-out allocated memory");
```

Comment:
Now that I think more about it, it isn't really feasible because by clearing the heap we mess with data allocated by the garbage collector for its own use.